### PR TITLE
fix set_transform link docs

### PR DESCRIPTION
**Pull request by diegulio**
# What does this PR do?

This PR simply fix a link to set_transform documentation. 

Fixes # (issue)

Broken link to set_transform docs

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu  @MKhalusova

### CODE REVIEW

File: docs/source/en/preprocessing.md [hunk 1]
```diff
@@ -412,7 +412,7 @@ If you wish to normalize images as a part of the augmentation transformation, us
 and `image_processor.image_std` values.
 </Tip>
 
-3. Then use 🤗 Datasets [`set_transform`](https://huggingface.co/docs/datasets/process.html#format-transform) to apply the transforms on the fly:
+3. Then use 🤗 Datasets [`set_transform`](https://huggingface.co/docs/datasets/v2.14.5/en/package_reference/main_classes#datasets.Dataset.set_transform) to apply the transforms on the fly:
```
Comment (ArthurZucker) [inline docs/source/en/preprocessing.md:416]:
```suggestion
3. Then use 🤗 Datasets [`set_transform`](https://huggingface.co/docs/datasets/package_reference/main_classes#datasets.Dataset.set_transform) to apply the transforms on the fly:
```

**Review by ArthurZucker — COMMENTED** [review]
Thanks but let's link to the up to date function!

File: docs/source/en/preprocessing.md [hunk 2]
```diff
@@ -412,7 +412,7 @@ If you wish to normalize images as a part of the augmentation transformation, us
...
```
Comment (stevhliu) [inline docs/source/en/preprocessing.md:415]: Lets make it even simpler by using the doc-builders syntax for referencing functions from other libraries :)

```suggestion
3. Then use 🤗 Datasets [`~datasets.Dataset.set_transform`] to apply the transforms on the fly:
```

**Review by stevhliu — APPROVED** [review]

File: docs/source/en/preprocessing.md (hunk 2 above)
Comment (diegulio) [reply docs/source/en/preprocessing.md]: thanks! @stevhliu

**Review by ArthurZucker — APPROVED** [review]
Nice thanks for the update

**Pull request merged**